### PR TITLE
Add swagger UI

### PR DESCRIPTION
**Pull request by Koshmaar**

### CODE REVIEW

File: tomcat-job-type/job-template.Dockerfile [hunk 1]
```diff
@@ -19,10 +21,21 @@ RUN apk add \
...
+# This time we have to unzip it manually, so that the folder is ready for the addition of swagger files.
+RUN unzip /tmp/app.war -d "${CATALINA_HOME}/webapps/pub#job#{{ manifest.name }}#{{ manifest.version }}/"
+
+# Swagger files are served from the same path.
+RUN cp -rv ${CATALINA_HOME}/webapps/swagger-ui/* "${CATALINA_HOME}/webapps/pub#job#{{ manifest.name }}#{{ manifest.version }}/"
+
+# Put the root.war under public url so that /swagger endpoint of Swagger servlet can be accessed by swagger-ui, and point it.
+RUN unzip -o "${CATALINA_HOME}/webapps/ROOT.war" -d "${CATALINA_HOME}/webapps/pub#job#{{ manifest.name }}#{{ manifest.version }}/"
+RUN sed -i 's/job_name/{{ manifest.name }}/' "${CATALINA_HOME}/webapps/pub#job#{{ manifest.name }}#{{ manifest.version }}/swagger-initializer.js"
+RUN sed -i 's/job_version/{{ manifest.version }}/' "${CATALINA_HOME}/webapps/pub#job#{{ manifest.name }}#{{ manifest.version }}/swagger-initializer.js"
```
Comment (iszulcdeepsense) [inline tomcat-job-type/job-template.Dockerfile:38]: Maybe you could merge it into one `RUN ... && ... && ...` command to reduce number of Docker layers

File: tomcat-job-type/app/src/main/java/Swagger.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                                      title: ValidationError
+                                  securitySchemes:
+                                    racetrackAuth:
+                                      type: apiKey
+                                      in: header
+                                      name: X-Racetrack-Auth
+                                security:
+                                  - racetrackAuth: []
+                                """;
+        String JobName = System.getenv("JOB_NAME");
```
Comment (iszulcdeepsense) [inline tomcat-job-type/app/src/main/java/Swagger.java:149]: Indentation looks odd here. I think it's under-indented

**Review by iszulcdeepsense — APPROVED** [review]

File: tomcat-job-type/job-template.Dockerfile (hunk 1 above)
Comment (Koshmaar) [reply tomcat-job-type/job-template.Dockerfile:38]: Good point.

File: tomcat-job-type/app/src/main/java/Swagger.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
+import jakarta.servlet.ServletException;
+import jakarta.servlet.annotation.WebServlet;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+
+@WebServlet(urlPatterns = {"/swagger"})
```
[the remark this answers is not in the document]
Comment (Koshmaar) [reply tomcat-job-type/app/src/main/java/Swagger.java:9]: Very good point.

File: tomcat-job-type/app/src/main/java/Swagger.java (hunk 2 above)
Comment (Koshmaar) [reply tomcat-job-type/app/src/main/java/Swagger.java:149]: Actually the swagger docstring was overindented.

**Pull request merged**